packages/types/src/chain/parachain/EvmParachain.ts

```diff
@@ -60,9 +62,19 @@ export class EvmParachain extends Parachain {
     this.id = id ?? 0;
     this.rpc = rpc ?? '';
     this.isEvmSigner = isEvmSigner;
+    this.ws = others.ws;
   }
 
   getViemChain(): Chain {
     return getViemChain(this);
   }
+
+  setWs(ws: string[]): void {
+    console.log('EvmParachain setWs called with:', ws);
+    this.ws = ws;
+  }
+
+  setRpc(rpc: string): void {
+    this.rpc = rpc;
+  }
```

Comment:
I would add similar methods as we have for Asset classes.

```suggestion
  copyWith(params: Partial<EvmParachainConstructorParams>): EvmParachain {
    return new EvmParachain({
      ...this,
      ...params,
    });
  }
```